coderd/database/dbmem/dbmem.go

```diff
@@ -9498,6 +9498,24 @@ func (q *FakeQuerier) ListWorkspaceAgentPortShares(_ context.Context, workspaceI
 	return shares, nil
 }
 
+func (q *FakeQuerier) MarkAllInboxNotificationsAsRead(_ context.Context, arg database.MarkAllInboxNotificationsAsReadParams) error {
```

Comment:
checking: we do not plan to _un-read_ the message?

Reply:
So there's two endpoints : 
- One which targets a specific notification (by id) - on this one we'll let user read or read it.
- One which targets all not-read notifications - on this one we'll just let user mark all as read - marking everything as unread would be super noisy.